### PR TITLE
ingress lib support for pydantic v2

### PR DESCRIPTION
**Pull request by PietroPasotti**
This PR makes the ingress library compatible with pydantic v1 and pydantic v2.

### CODE REVIEW

File: lib/charms/traefik_k8s/v2/ingress.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 ```shell
 cd some-charm
-charmcraft fetch-lib charms.traefik_k8s.v2.ingress
+charmcraft fetch-lib charms.traefik_k8s.v1.ingress
```
Comment (mmkay) [inline lib/charms/traefik_k8s/v2/ingress.py:16]: This should stay at v2, I think?

File: lib/charms/traefik_k8s/v2/ingress.py [hunk 2]
```diff
@@ -72,69 +72,131 @@ def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent):
...
+            if nest_under:
+                return cls.model_validate(json.loads(databag[nest_under]))
 
-        if databag is None:
-            databag = {}
+            try:
+                data = {k: json.loads(v) for k, v in databag.items() if k not in BUILTIN_JUJU_KEYS}
+            except json.JSONDecodeError as e:
+                msg = f"invalid databag contents: expecting json. {databag}"
+                logger.error(msg)
```
Comment (mmkay) [inline lib/charms/traefik_k8s/v2/ingress.py:163]: Most likely should be `log` - logger in this lib is under a `log` reference.

**Review by mmkay — APPROVED** [review]
There's a couple of failing pipelines that might need updating, but overall I think it's better than creating a `v3` ingress just for pydantic upgrade. A couple of minor comments left.

**Comment by PietroPasotti** [comment]
note the unittest failures are due to what's being fixed in #287

**Pull request merged**